framework-cloud/framework-openshift-operator/src/main/java/org/kie/cloud/openshift/operator/scenario/ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenarioImpl.java

```diff
@@ -0,0 +1,198 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.cloud.openshift.operator.scenario;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import cz.xtf.openshift.OpenShiftBinaryClient;
+import cz.xtf.wait.SimpleWaiter;
+import org.kie.cloud.api.deployment.ControllerDeployment;
+import org.kie.cloud.api.deployment.DatabaseDeployment;
+import org.kie.cloud.api.deployment.Deployment;
+import org.kie.cloud.api.deployment.KieServerDeployment;
+import org.kie.cloud.api.deployment.SmartRouterDeployment;
+import org.kie.cloud.api.deployment.WorkbenchDeployment;
+import org.kie.cloud.api.scenario.ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenario;
+import org.kie.cloud.common.provider.KieServerControllerClientProvider;
+import org.kie.cloud.openshift.deployment.DatabaseDeploymentImpl;
+import org.kie.cloud.openshift.deployment.KieServerDeploymentImpl;
+import org.kie.cloud.openshift.deployment.SmartRouterDeploymentImpl;
+import org.kie.cloud.openshift.deployment.WorkbenchRuntimeDeploymentImpl;
+import org.kie.cloud.openshift.operator.resources.OpenShiftResource;
+import org.kie.cloud.openshift.resource.Project;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenarioImpl extends OpenShiftOperatorScenario implements ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenario {
+
+    private WorkbenchDeployment workbenchRuntimeDeployment;
+    private SmartRouterDeployment smartRouterDeployment;
+    private KieServerDeploymentImpl kieServerOneDeployment;
+    private KieServerDeploymentImpl kieServerTwoDeployment;
+    private DatabaseDeploymentImpl databaseOneDeployment;
+    private DatabaseDeploymentImpl databaseTwoDeployment;
+
+    private static final Logger logger = LoggerFactory.getLogger(ClusteredWorkbenchRuntimeSmartRouterTwoKieServersTwoDatabasesScenarioImpl.class);
+
+    @Override
+    public void deploy() {
+        super.deploy();
+
+        // deploy application
+        logger.info("Creating application from " + OpenShiftResource.CLUSTERED_WORKBENCH_KIE_SERVER_DATABASE_PERSISTENT.getResourceUrl().toString());
+        OpenShiftBinaryClient.getInstance().executeCommand("Deployment failed.", "create", "-f", OpenShiftResource.CLUSTERED_WORKBENCH_KIE_SERVER_DATABASE_PERSISTENT.getResourceUrl().toString());
+
+        // Wait until deployment is created
+        // Temporary, replace by something better.
+        try {
+            new SimpleWaiter(() -> project.getOpenShiftUtil().getServices().size() == 9).timeout(TimeUnit.MINUTES, 1).execute();
```

Comment:
Is timeout big enough for deploy?

Reply:
It is actually a waiting loop for Operator to initialize scenario (create all services, routes....), it is checking the available configuration every 10 seconds and initialization itself is a matter of seconds.

Reply:
ok, cool